infrastructure/src/hasura/parameters/index.ts

```diff
@@ -1,13 +1,25 @@
-import * as cdk from 'aws-cdk-lib'
-import { dev } from './dev'
-import { production } from './production'
-import { staging } from './staging'
-import { ContextEnv } from 'common/types'
+import * as cdk from "aws-cdk-lib";
+import { dev } from "./dev";
+import { production } from "./production";
+import { staging } from "./staging";
+import { ContextEnv } from "common/types";
+
+type UndefinedKeys<T> = {
+  [P in keyof T]: undefined extends T[P] ? P : never;
+}[keyof T];
+
+// @ANCHOR Makes the key optional if the value of this key has undefined type in the declaration
+type OptionalUndefined<T> = Partial<Pick<T, UndefinedKeys<T>>> &
+  Omit<T, UndefinedKeys<T>>;
+
+type Envs = OptionalUndefined<ReturnType<typeof staging>["envs"]> &
+  OptionalUndefined<ReturnType<typeof dev>["envs"]> &
+  OptionalUndefined<ReturnType<typeof production>["envs"]>;
 
 export const parameters = (params: {
-  environment: ContextEnv
-  hostedZone: cdk.aws_route53.IHostedZone
+  environment: ContextEnv;
```

Comment:
@igorosip0v the presence of `"` and `;` looks like code styling inconsistency.

Reply:
@gvidon Are we using manual formatting here?

There is `formatOnSave` enabled in `.vscode` and prettier formats with default settings, I suppose, because there is no any settings for prettier, as I can see.

Reply:
Updated

Reply:
@igorosip0v it should be prettier I think. Isn't it working?